### DISCUSSION TITLE
02. PyTorch Neural Network Classification (timestamp: 12:34:46 - challenge: minimise the loss on test data on CIrcleModelV3)

### DISCUSSION THREAD

**Discussion by tiff78210241911** [mrdbourke/pytorch-deep-learning#1245] [Q&A]
Hi, I'm attempting to improve my model to have a higher accuracy than 80% on test data.  I have added two more layers (2 -> 5 ->10 -> 10 -> 5 -> 1) so 5 layers in total and changed the learning rate to 0.01. However, neither the loss nor the accuracy is improving. I'm not sure why. Any feedback would be greatly appreciated. Thanks!

<img width="800" height="500" alt="image" src="https://github.com/user-attachments/assets/06b359a9-acf9-441f-9961-7f075dd08211" />

<img width="580" height="122" alt="image" src="https://github.com/user-attachments/assets/89bfb017-5d09-4c45-89ab-f9efcfc2cab4" />

```
torch.manual_seed(42)
torch.cuda.manual_seed(42)

epochs = 1000
# put all data to target devices
X_train, y_train = X_train.to(device), y_train.to(device)
X_test, y_test = X_test.to(device), y_test.to(device)

for epoch in range(epochs):
  ### training
  model_4.train()

  # 1. forward pass
  y_logits = model_4(X_train).squeeze()
  y_preds = torch.round(torch.sigmoid(y_logits))

  # 2. calculate the loss
  loss = loss_fn(y_logits, y_train)
  acc = accurancy_fn(y_true=y_train,
                     y_pred=y_preds)
  
  # 3. optimiser zero grad
  optimsier.zero_grad()

  # 4. loss backward
  loss.backward()

  # 5. optimer step
  optimiser.step()

  ### testing
  model_4.eval()

  with torch.inference_mode():
    # 1. forward pass
    test_logits = model_4(X_test).squeeze()
    test_preds = torch.round(torch.sigmoid(test_logits))

    # 2. calculate the loss
    test_loss = loss_fn(test_logits, y_test)
    test_acc = accurancy_fn(y_true=y_test,
                            y_pred=test_preds)
  
  # print out some stats
  if epoch % 100 == 0:
    print(f"Epoch: {epoch}  | Loss: {loss:.5f}, Acc : {acc:.2f}% | Test loss: {test_loss:.5f}, Test acc: {test_acc:.2f}%")
```

<img width="917" height="239" alt="image" src="https://github.com/user-attachments/assets/0d39790a-81cd-4260-941b-8f19cab445a8" />
